src/mame/drivers/seta.cpp

```diff
@@ -3026,16 +3026,90 @@ void kiwame_state::kiwame_map(address_map &map)
                         Thunder & Lightning / Wit's
 ***************************************************************************/
 
+/* Protection only present in thunderl set.
+   Implemented using a registered PALCE16V8H: TL-9. Main CPU performs several writtings to the address space 
+   mapped to the PAL to save a value into PAL registers. Eventually, CPU reads back that value and performs some checkings.
+   If the value is not the proper one, a soft reset is done.
+   Address during writting operation is mapped in the following way:
+
+   A2  -> I2
+   A3  -> I3
+   A6  -> I4
+   A8  -> I5
+   A11 -> I6
+   A13 -> I7
+   A15 -> I8
+   A16 -> I9
+
+   Data sent by the CPU during these writting operations is not used to calculate the protection register value, only some
+   address lines are used to compute the value, as shown above.
+   Every write operation done to the PAL discards the previous stored value in the registers and stores a new computed value,
+   following the logic equations programmed in the PAL.
+
+   (I1 = CLK : pulses when accessing to writting handler and acts as clock for internal latches in PAL)
+   (I11 = /OE : asserted when accessing to reading handler. Sub-address used for reading here is no used to compute the result value
+
+   I19  -> D0
+   I18  -> D1
+   I17  -> D2
+   I16  -> D3
+   I15  -> D4
+   I14  -> D5
+   I13  -> D6
+   I12  -> D7
+*/
```

Comment:
Please reformat this comment to not have excessively long lines – use the row of asterisks in the heading comment above it as a guide.